pkg/scheduler/scheduler_test.go

```diff
@@ -808,20 +807,30 @@ func setupTestScheduler(queuedPodStore *clientcache.FIFO, scache internalcache.C
 		return true, b, nil
 	})
 
-	fwk, _ := st.NewFramework(
-		fns,
-		frameworkruntime.WithClientSet(client),
-		frameworkruntime.WithInformerFactory(informerFactory),
-		frameworkruntime.WithPodNominator(internalqueue.NewPodNominator()),
-	)
-	prof := &profile.Profile{
-		Framework: fwk,
-		Recorder:  &events.FakeRecorder{},
-		Name:      testSchedulerName,
-	}
+	var fwk framework.Framework
 	if broadcaster != nil {
-		prof.Recorder = broadcaster.NewRecorder(scheme.Scheme, testSchedulerName)
+		recorder := broadcaster.NewRecorder(scheme.Scheme, testSchedulerName)
+		fwk, _ = st.NewFramework(
+			fns,
+			frameworkruntime.WithClientSet(client),
+			frameworkruntime.WithEventRecorder(recorder),
```

Comment:
why not call `NewFramework` after the conditionals with the appropriate recorder?